### PR TITLE
[Merged by Bors] - feat port: RingTheory.Coprime.Basic

### PR DESCRIPTION
**Pull request by ChrisHughes24**
a95b16cbade0f938fc24abd05412bde1e84bab9b

Turns out there are actually some more imports needed to port this file which were imported via `tactic.ring` in Lean3.

### CODE REVIEW

File: Mathlib/RingTheory/Coprime/Basic.lean [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+import Mathlib.Tactic.Ring
+import Mathlib.GroupTheory.GroupAction.Units
+import Mathlib.Algebra.Ring.Divisibility
+
+/-!
+# Coprime elements of a ring
+
+## Main definitions
+
+* `isCoprime x y`: that `x` and `y` are coprime, defined to be the existence of `a` and `b` such
```
Comment (Ruben-VandeVelde) [inline Mathlib/RingTheory/Coprime/Basic.lean:15]:
```suggestion
* `IsCoprime x y`: that `x` and `y` are coprime, defined to be the existence of `a` and `b` such
```

File: Mathlib/RingTheory/Coprime/Basic.lean [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+  ⟨fun ⟨a, b, H⟩ => isUnit_of_mul_eq_one x b <| by rwa [mul_zero, zero_add, mul_comm] at H, fun H =>
+    let ⟨b, hb⟩ := isUnit_iff_exists_inv'.1 H
+    ⟨1, b, by rwa [one_mul, zero_add]⟩⟩
+#align is_coprime_zero_left isCoprime_zero_left
+
+theorem isCoprime_zero_right : IsCoprime x 0 ↔ IsUnit x :=
+  isCoprime_comm.trans isCoprime_zero_left
+#align is_coprime_zero_right isCoprime_zero_right
+
+theorem not_coprime_zero_zero [Nontrivial R] : ¬IsCoprime (0 : R) 0 :=
```
Comment (semorrison) [inline Mathlib/RingTheory/Coprime/Basic.lean:68]:
```suggestion
theorem not_isCoprime_zero_zero [Nontrivial R] : ¬IsCoprime (0 : R) 0 :=
```

Comment (semorrison) [reply Mathlib/RingTheory/Coprime/Basic.lean:68]: Also a few below. These aren't essential, obviously, but this is a cheap moment to rename a few things.

File: Mathlib/RingTheory/Coprime/Basic.lean [hunk 3]
```diff
@@ -0,0 +1,391 @@
...
+theorem isCoprime_group_smul_left : IsCoprime (x • y) z ↔ IsCoprime y z :=
+  ⟨fun ⟨a, b, h⟩ => ⟨x • a, b, by rwa [smul_mul_assoc, ← mul_smul_comm]⟩, fun ⟨a, b, h⟩ =>
+    ⟨x⁻¹ • a, b, by rwa [smul_mul_smul, inv_mul_self, one_smul]⟩⟩
+#align is_coprime_group_smul_left isCoprime_group_smul_left
+
+theorem isCoprime_group_smul_right : IsCoprime y (x • z) ↔ IsCoprime y z :=
+  isCoprime_comm.trans <| (isCoprime_group_smul_left x z y).trans isCoprime_comm
+#align is_coprime_group_smul_right isCoprime_group_smul_right
+
+theorem is_coprime_group_smul : IsCoprime (x • y) (x • z) ↔ IsCoprime y z :=
```
Comment (semorrison) [inline Mathlib/RingTheory/Coprime/Basic.lean:236]:
```suggestion
theorem isCoprime_group_smul : IsCoprime (x • y) (x • z) ↔ IsCoprime y z :=
```

**Comment by semorrison** [comment]
bors d+

**Comment by ChrisHughes24** [comment]
bors merge

**Pull request closed without merging**